### DISCUSSION TITLE
Question: Wait instead of cancelling if it is executing?

### DISCUSSION THREAD

**Discussion by ghost** [mhenrixon/sidekiq-unique-jobs#656] [Q&A] [answered]
Hi,

I am sorry in advance if it is inappropriate to ask questions here. I was wondering if anyone could point me in the right direction to create this behavior:

Let's say I queue MyJob(param=123). I cannot re-queue MyJob(param=123) until it is executing(in-progress). (Behavior of until_executing). The edge case here is while MyJob(param=123) is executing, it should be able to queue again the same job, but it should wait until the first MyJob(param=123) is done before running the second instance with the same param.

According to the doc, there is a on_conflict: :reschedule, however it's not exactly this behavior I am looking for. The correct behavior would be :

:reject if job with same param is already queued
:reschedule if job with same param is running and there is not another one in the queue.

Thanks in advance :)

**Comment by mhenrixon** [comment, marked as answer]
This would do the trick.

```ruby
sidekiq_options on_conflict: { client: :reject, server: :reschedule } 
```

**Reply by ghost** [reply]
Thanks for the quick reply @mhenrixon 

Do I need to specify de lock method? Could you please explain it a little bit ? :)

**Reply by mhenrixon** [reply]
It assums you are using `lock: :until_and_while_executing`. For example, `:until_executed` doesn't use a different lock for the server process so it wouldn't unlock for pushes when your worker performs the task.
